platform/api/aliyun/api.go

```diff
@@ -168,7 +169,54 @@ func (a *API) ImportImage(format, bucket, object, image_size, device, name, desc
 		return "", fmt.Errorf("importing image: %v", err)
 	}
 
-	return response.ImageId, nil
+	return a.waitForImportImageTask(response)
+}
+
+func (a *API) waitForImportImageTask(importImageResponse *ecs.ImportImageResponse) (string, error) {
+	importDone := func(taskId string) (bool, error) {
+		request := ecs.CreateDescribeTasksRequest()
+		request.TaskIds = taskId
+		res, err := a.ecs.DescribeTasks(request)
+		if err != nil {
+			return false, err
+		}
+
+		if len(res.TaskSet.Task) != 1 {
+			return false, fmt.Errorf("expected result about one task, got %v", res.TaskSet.Task)
+		}
+
+		switch res.TaskSet.Task[0].TaskStatus {
+		case "Finished":
+			return true, nil
+		case "Processing":
+			return false, nil
+		case "Waiting":
+			return false, nil
+		case "Deleted":
+			return false, fmt.Errorf("task %v was deleted", taskId)
+		case "Paused":
+			return false, fmt.Errorf("task %v was paused", taskId)
+		case "Failed":
+			return false, fmt.Errorf("task %v failed", taskId)
+		default:
+			return false, fmt.Errorf("unexpected status for task %v: %v", taskId, res.TaskSet.Task[0].TaskStatus)
+		}
+	}
+
+	for {
+		var done bool
+		var err error
+		done, err = importDone(importImageResponse.TaskId)
```

Comment:
nonblocking:
Why not use `:=` here?

Reply:
So, this code is actually based on AWS' `finishSnapshotTask()`, which also does this. Though in their case, they have to use `=` because of variable in an outer scope. We don't, so I switched this one here to `:=`.